e2e/test-10-self-signed-tls.bats

```diff
@@ -0,0 +1,580 @@
+#!/usr/bin/env bats
+
+load "lib/utils"
+load "lib/detik"
+load "lib/k8up"
+
+# shellcheck disable=SC2034
+DETIK_CLIENT_NAME="kubectl"
+# shellcheck disable=SC2034
+DETIK_CLIENT_NAMESPACE="k8up-e2e-subject"
+# shellcheck disable=SC2034
+DEBUG_DETIK="true"
+
+### Start backup section
+
+@test "Given a PVC, When creating a Backup (TLS) of an app, Then expect Restic repository - using self-signed issuer" {
```

Comment:
Can you reduce this to 1 TLS and 1 mTLS test? This file contains more end-to-end tests than the rest of k8up combined and takes almost 1h to run. This is pretty hard to maintain.

E2e tests are usually not used to test each permutation of every possible scenario. That's something for unit and integration tests. E2e tests should be few and pretty broad. (Yes I know that there aren't that much of those either in K8up...)

Reply:
I'd probably go with something like `test-04-restore.bats` does. A backup and a restore in one single e2e test. (IMHO test-04-restore.bats probably makes the test-03-backup.bats test more or less redundant anyway. But that's a problem for another issue :) )

Reply:
I removed the multiply e2e tests and created an integration for them.